js/src/logger.ts

```diff
@@ -3215,6 +3279,36 @@ export class SpanImpl implements Span {
     let sanitizedAndInternalData = { ...internalData };
     mergeDicts(sanitizedAndInternalData, sanitized);
 
+    const serializableInternalData: typeof sanitizedAndInternalData = {};
+    const lazyInternalData: Record<string, LazyValue<unknown>> = {};
+
+    for (const [key, value] of Object.entries(sanitizedAndInternalData) as [
+      keyof typeof sanitizedAndInternalData,
+      any,
+    ][]) {
+      if (value instanceof BraintrustStream) {
+        const streamCopy = value.copy();
+        lazyInternalData[key] = new LazyValue(async () => {
+          return await new Promise((resolve) => {
+            streamCopy
+              .toReadableStream()
+              .pipeThrough(createFinalValuePassThroughStream(resolve))
+              .pipeTo(devNullWritableStream());
+          });
+        });
+      } else if (value instanceof ReadableStream) {
+        lazyInternalData[key] = new LazyValue(async () => {
+          return await new Promise((resolve) => {
+            value
+              .pipeThrough(createFinalValuePassThroughStream(resolve))
```

Comment:
Is this safe? What if someone logs a `ReadableStream` of their own somewhere which doesn't contain one of the types admissible by `createFinalValuePassThroughStream`? Is it possible to check for this sort-of compatibility up front (or force people to use `BraintrustStream` everywhere)? Otherwise it seems like the `createFinalValuePassThroughStream` code will just assume it's a `BraintrustStreamChunk` if not string or `Uint8Array`.

Reply:
Yep. If it's a braintrust stream, it parses the value. Otherwise it just forwards along the text.

Reply:
I was just thinking that the `ReadableStream` type signature seems to allow any type of data (it actually becomes `any` in this code), but we are assuming in the stream transformer that it is of type `string | Uint8Array | BraintrustStreamChunk`. Is that assumption necessarily safe, or maybe it's uncommon-enough that a weird crash is okay?

I'm kind-of just wondering why we need to handle the case of a bare `ReadableStream` in the first place?